python/sparkdl/graph/input.py

```diff
@@ -0,0 +1,232 @@
+
+    @classmethod
+    def _new_obj_internal(cls):
+        # pylint: disable=attribute-defined-outside-init
+        obj = object.__new__(cls)
+        # TODO: for (de-)serialization, the class should correspond to a ProtocolBuffer definition.
+        obj.graph_def = None
+        obj.input_tensor_name_from_signature = None
+        obj.output_tensor_name_from_signature = None
+        return obj
+
+    @classmethod
+    def fromGraph(cls, graph, sess, feed_names, fetch_names):
+        """
+        Construct a TFInputGraphBuilder from a in memory tf.Graph object
+        """
+        assert isinstance(graph, tf.Graph), \
+            ('expect tf.Graph type but got', type(graph))
+
+        def import_graph_fn(_sess):
+            assert _sess == sess, 'must have the same session'
+            return _GinBuilderInfo()
+
+        return _GinBuilder(import_graph_fn, sess, graph).build(feed_names, fetch_names)
+
+    @classmethod
+    def fromGraphDef(cls, graph_def, feed_names, fetch_names):
+        """
+        Construct a TFInputGraphBuilder from a tf.GraphDef object
+        """
+        assert isinstance(graph_def, tf.GraphDef), \
+            ('expect tf.GraphDef type but got', type(graph_def))
+
+        def import_graph_fn(sess):
+            with sess.as_default():
+                tf.import_graph_def(graph_def, name='')
+            return _GinBuilderInfo()
+
+        return _GinBuilder(import_graph_fn).build(feed_names, fetch_names)
+
+    @classmethod
+    def fromCheckpoint(cls, checkpoint_dir, feed_names, fetch_names):
+        return cls._from_checkpoint_impl(checkpoint_dir,
+                                         signature_def_key=None,
+                                         feed_names=feed_names, fetch_names=fetch_names)
+
+    @classmethod
+    def fromCheckpointWithSignature(cls, checkpoint_dir, signature_def_key):
+        assert signature_def_key is not None
+        return cls._from_checkpoint_impl(checkpoint_dir,
+                                         signature_def_key,
+                                         feed_names=None, fetch_names=None)
+
+    @classmethod
+    def fromSavedModel(cls, saved_model_dir, tag_set, feed_names, fetch_names):
+        return cls._from_saved_model_impl(saved_model_dir, tag_set,
+                                          signature_def_key=None,
+                                          feed_names=feed_names, fetch_names=fetch_names)
+
+    @classmethod
+    def fromSavedModelWithSignature(cls, saved_model_dir, tag_set, signature_def_key):
+        assert signature_def_key is not None
+        return cls._from_saved_model_impl(saved_model_dir, tag_set,
+                                          signature_def_key=signature_def_key,
+                                          feed_names=None, fetch_names=None)
+
+    @classmethod
+    def _from_checkpoint_impl(cls,
+                              checkpoint_dir,
```

Comment:
use consistent style for the python function declaration.

Reply:
YAPF'ed function declaration